subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/IndexPageGenerator.java

```diff
@@ -118,4 +122,16 @@ public void render(final ResultsStore store, Writer writer) throws IOException {
             endAll();
         }};
     }
+
+    private List<? extends PerformanceTestExecution> filterForRequestedCommit(List<? extends PerformanceTestExecution> results) {
+        if (commitId == null) {
+            return results;
+        }
+        for (PerformanceTestExecution execution : results) {
+            if (execution.getVcsCommits().contains(commitId)) {
```

Comment:
Could be more efficient by putting this logic in the database query (i.e. passing the parameter into the store). Not a big deal though, we're talking a few dozen entries here.